java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -1,134 +1,150 @@
-        return this;
-    }
-
-    public SauceOptions withLinux() {
-        operatingSystem = "Linux";
-        return this;
-    }
-
-    public SauceOptions withWindows10() {
-        operatingSystem = "windows 10";
-        return this;
-    }
-    public SauceOptions withWindows8_1() {
-        operatingSystem = "Windows 8.1";
-        return this;
-    }
-    public SauceOptions withWindows8() {
-        operatingSystem = "Windows 8";
-        return this;
-    }
-
-    public SauceOptions withWindows7() {
-        operatingSystem = "Windows 7";
-        return this;
-    }
-
-    public SauceOptions withEdge() {
-        browserName = "Edge";
-        browserVersion = "18.17763";
-        return this;
-    }
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 
-    //TODO notice the duplication below with edge.
-    //Maybe could be moved to a separate class so we can do withEdge().16_16299();
-    //Or withEdge().version(EdgeVersion.16_16299);
-    public SauceOptions withEdge18() {
-        withEdge();
-        browserVersion = "18.17763";
-        return this;
-    }
-    public SauceOptions withEdge17() {
-        withEdge();
-        browserVersion = "17.17134";
-        return this;
-    }
-    public SauceOptions withEdge16() {
-        withEdge();
-        browserVersion = "16.16299";
-        return this;
-    }
-
-    public SauceOptions withEdge15() {
-        withEdge();
-        browserVersion = "15.15063";
-        return this;
-    }
-
-    public SauceOptions withEdge14() {
-        withEdge();
-        browserVersion = "14.14393";
-        return this;
-    }
-
-    public SauceOptions withEdge13() {
-        withEdge();
-        browserVersion = "13.10586";
-        return this;
-    }
-
-    public SauceOptions withFirefox() {
-        browserName = "Firefox";
-        return this;
-    }
-
-    public SauceOptions withIE(IEVersion version) {
-        this.browserVersion = version.getVersion();
-        return this;
-    }
-
-    public SauceOptions withIE(String version) {
-        browserName = "IE";
-        browserVersion = version;
-        return this;
-    }
-
-    public SauceOptions withIE() {
-        browserName = "IE";
-        browserVersion = "latest";
-        return this;
-    }
+public class SauceOptions {
+    @Getter private MutableCapabilities seleniumCapabilities;
 
-    public SauceOptions withMac(MacVersion macVersion) {
-        operatingSystem = macVersion.label;
-        browserName = "Safari";
-        return this;
+    // w3c Settings
+    @Getter @Setter private String browserName = "chrome";
+    @Getter @Setter private String browserVersion = "latest";
+    @Getter @Setter private String platformName = "Windows 10";
+    @Getter @Setter private String pageLoadStrategy;
+    @Getter @Setter private Boolean acceptInsecureCerts = null;
```

Comment:
The reasons for this...
The lombok getter for a `boolean` type is `is` instead of `get`, and handling that with reflection was an extra hassle. Additionally, the default value for a `boolean` is `false`, which would override the actual desired default based on w3c, which I do not want to do. If it is `null`, I can just skip it when creating `toCapabilities`.

Reply:
You can avoid the `is` prefix and just get straight getters using the Lombok `lombok.getter.noIsPrefix` configuration option. I'd strongly suggest using `java.beans.Introspector` as the entry-point for your reflection-based magic: that thing's designed for handling "properties" of beans nicely.